backend/src/main/java/com/redhat/cloud/notifications/events/FromCamelHistoryFiller.java

```diff
@@ -54,6 +78,76 @@
                 });
     }
 
+    private void reinjectIfNeeded(Map<String, Object> payload) {
+        if (!payload.containsKey("successful") || !((Boolean)payload.get("successful"))) {
+            String historyId = (String) payload.get("historyId");
+            log.infof("Event with id %s was not successful, resubmitting for further processing", historyId);
+
+            Event event = new Event();
+
+            Uni<Endpoint> endpointUni = notificationResources.getEndpointForHistoryId(historyId);
+            endpointUni.onItem().transform(ep -> {
+
+                // TODO augment with details from Endpoint and original event
+                event.setPayload(payload);
+
+                // Save the original id, as we may need it in the future.
+                Map<String,String> context = new HashMap<>();
+                context.put("original-id", historyId);
+                context.put("failed-integration", ep.getName());
+
+                Action action = new Action(
+                        "platform",
+                        "notifications",
+                        "integration_failed",
+                        LocalDateTime.now(),
+                        ep.getAccountId(),
+                        context, // context
+                        Collections.singletonList(event),
+                        "v1.0.0",
+                        Collections.emptyList()
+                    );
+                try {
+                    String ser = serializeAction(action);
+                    // Add the message id in Kafka header for the de-duplicator
+                    Message<String> message = buildMessageWithId(ser);
+                    emitter.send(message);
+                    return Uni.createFrom().voidItem();
+                } catch (IOException e) {
+                    e.printStackTrace();
+                    return Uni.createFrom().failure(e);
+                }
+
+            })
+            .subscribe()
+                    .with(x -> x.onItem().invoke(item -> System.out.println(item))); // This triggers the actual work
+
+
+        }
+    }
+
+    // Blindly copied from -gw . Perhaps put this into Schema project?
```

Comment:
https://github.com/RedHatInsights/insights-schemas-java/blob/master/insights-notification-schemas-java/src/main/java/com/redhat/cloud/notifications/ingress/Encoder.java#L17

Reply:
Nice, but it could be a static method _nitpick_